pennylane/resources/first_quantization.py

```diff
@@ -0,0 +1,405 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed for estimating the number of logical qubits and
+non-Clifford gates for quantum algorithms in first quantization using a plane-wave basis.
+"""
+# pylint: disable= too-many-arguments
+from pennylane import numpy as np
+
+
+def _cost_qrom_fq(lz):
+    r"""Return the minimum number of Toffoli gates needed for erasing the output of a QROM.
+
+    Args:
+        lz (int): sum of the atomic numbers of nuclei
+
+    Returns:
+        int: the minimum cost of erasing the output of a QROM
+
+    **Example**
+
+    >>> _cost_qrom_min(100)
+    21
+    """
+    k_f = np.floor(np.log2(lz) / 2)
+    k_c = np.ceil(np.log2(lz) / 2)
+
+    cost_f = int(2**k_f + np.ceil(2 ** (-1 * k_f) * lz))
+    cost_c = int(2**k_c + np.ceil(2 ** (-1 * k_c) * lz))
+
+    return min(cost_f, cost_c)
+
+
+def unitary_cost_fq(n, eta, omega, error, lamb, br=7, charge=0):
+    r"""Return the number of Toffoli gates needed to implement the qubitization unitary operator.
+    ￼
+    The expression for computing the cost is taken from Eq. (125) of
```

Comment:
Is the expression given in `Theory`? If yes, we might mention it here.

Reply:
No, the theory explains some details of how we assign default values to different errors. I decided not to copy all the equations in the docs but just refer to them with exact equation numbers.